src/Common/tests/System/PlatformDetection.cs

```diff
@@ -408,5 +405,33 @@ public static bool IsNonZeroLowerBoundArraySupported
         // System.Security.Cryptography.Xml.XmlDsigXsltTransform.GetOutput() relies on XslCompiledTransform which relies
         // heavily on Reflection.Emit
         public static bool IsXmlDsigXsltTransformSupported => PlatformDetection.IsReflectionEmitSupported;
+
+        public static Range[] FrameworkRanges => new Range[]{
+          new Range(new Version(4, 7, 2500, 0), null, new Version(4, 7, 1)),
+          new Range(new Version(4, 6, 2000, 0), new Version(4, 7, 2090, 0), new Version(4, 7, 0)),
+          new Range(new Version(4, 6, 1500, 0), new Version(4, 6, 1999, 0), new Version(4, 6, 2)),
+          new Range(new Version(4, 6, 1000, 0), new Version(4, 6, 1499, 0), new Version(4, 6, 1)),
+          new Range(new Version(4, 6, 55, 0), new Version(4, 6, 999, 0), new Version(4, 6, 0)),
+          new Range(new Version(4, 0, 30319, 0), new Version(4, 0, 52313, 36313), new Version(4, 5, 2))
```

Comment:
This could also be done like:
```cs
if (version.BuildNumber >= 2500)
   return new Version(4, 7, 1);
if (version.BuildNumber >= 2000)
   return new Version(4, 7, 0);

   ...

}
```
but with the Range pattern if the build numbers/revision patterns changes we would be safe as we could keep adding ranges into the array to map to new versions. As for example 4.5.2 doesn't follow the incremental build pattern, so it is more accurate to follow the Range pattern.

Didn't add 4.5.1 ranges as it had too much releases and we want to unblock the Serialization PR ASAP. They can be added in a separate PR.